storage/src/mmr/mem.rs

```diff
@@ -90,31 +87,51 @@ impl<H: CHasher> Mmr<H> {
     }
 
     /// Add an element to the MMR and return its position in the MMR.
+    ///
+    /// The implementation initializes a new Hasher object with each call. For bulk additions,
+    /// consider using `add_with_hasher` to avoid this overhead.
     pub fn add(&mut self, element: &H::Digest) -> u64 {
+        self.add_with_hasher(element, &mut H::new())
+    }
+
+    /// Add an element to the MMR and return its position in the MMR.
+    ///
+    /// This version requires the caller to provide an already initialized hasher for reuse to
+    /// optimize bulk additions.
+    pub fn add_with_hasher(&mut self, element: &H::Digest, hasher: &mut H) -> u64 {
```

Comment:
By convention, we tend to put things needed to "do the work" before args. So, this would be: `self, hasher, element`

Reply:
changed